src/main/java/nerdhub/textilelib/mixin/client/MixinInGameHud.java

```diff
@@ -0,0 +1,89 @@
+package nerdhub.textilelib.mixin.client;
+
+import net.minecraft.client.MinecraftClient;
+import net.minecraft.client.gui.hud.*;
+
+import nerdhub.textilelib.event.client.render.hud.DrawHudCallback;
+import nerdhub.textilelib.event.client.render.hud.VanillaHudTypes;
+import org.spongepowered.asm.mixin.Final;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+
+@Mixin(InGameHud.class)
+public abstract class MixinInGameHud {
+
+    @Shadow
+    @Final
+    private MinecraftClient client;
+
+    @Shadow
+    @Final
+    private DebugHud debugHud;
+
+    @Shadow
+    @Final
+    private BossBarHud bossBarHud;
+
+    @Shadow
+    @Final
+    private SpectatorHud spectatorHud;
+
+    @Shadow
+    @Final
+    private SubtitlesHud subtitlesHud;
+
+    @Shadow
+    @Final
+    private ScoreboardHud scoreboardHud;
+
+    @Shadow
+    @Final
+    private ChatHud chatHud;
+
+    @Inject(at = @At(value = "INVOKE", target = "Lnet/minecraft/client/gui/hud/SpectatorHud;draw(F)V", shift = At.Shift.AFTER), method = "draw")
+    private void drawSpectatorHud(float deltaTime, CallbackInfo ci) {
+        this.client.getProfiler().swap("textilelib_render_hud_spectator");
```

Comment:
maybe use `textilelib:` for the debug profiler? imo that is a better format.

Reply:
After `textilelib:` should it be camel case instead of `_`?

Reply:
Since that is what MC uses

Reply:
good idea, yes